cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/UserPermissions.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2023 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet.response;
+
+import com.linkedin.cruisecontrol.servlet.parameters.CruiseControlParameters;
+import com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig;
+import com.linkedin.kafka.cruisecontrol.config.constants.WebServerConfig;
+import com.linkedin.kafka.cruisecontrol.servlet.UserPermissionsManager;
+import com.linkedin.kafka.cruisecontrol.servlet.UserRequestException;
+import com.linkedin.kafka.cruisecontrol.servlet.parameters.UserPermissionsParameters;
+import java.util.HashSet;
+import java.util.Set;
+import com.google.gson.Gson;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * - If security enabled for Cruise Control, then the roles will be the result which were assigned to the user:
+ *      e.g. ["ADMIN", "VIEWER"], with 200 HTTP status code.
+ * - If security disabled, then it will result a 400 HTTP status code, with the following message:
+ *      "Unable to retrieve privilege information for an unsecure connection."
+ */
+@JsonResponseClass
+public class UserPermissions extends AbstractCruiseControlResponse {
+
+    @JsonResponseField(required = false)
+    protected static final String ROLES = "roles";
```

Comment:
Just curious why is it protected instead of private?

Reply:
Will be made public due to integration tests